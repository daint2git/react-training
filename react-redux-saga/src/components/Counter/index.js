import React from 'react'
import { connect } from 'react-redux'

import { increment, incrementAsync, decrement } from 'redux/reducers/counter'

const Counter = ({ count, onIncrement, onIncrementAsync, onDecrement }) => (
  <div>
    <button type="button" onClick={onIncrement}>
      +
    </button>
    <button type="button" onClick={onIncrementAsync}>
      + (with async)
    </button>
    <button type="button" onClick={onDecrement}>
      -
    </button>
    <p>{count}</p>
  </div>
)

export default connect(
  state => ({ count: state.counter.count }),
  {
    onIncrement: increment,
    onIncrementAsync: incrementAsync,
    onDecrement: decrement,
  },
)(Counter)
