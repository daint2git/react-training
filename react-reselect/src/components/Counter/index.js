import React from 'react'
import { connect } from 'react-redux'

import { increment } from 'redux/reducers/counterReducer'

class Counter extends React.Component {
  componentDidMount() {
    const { increment } = this.props
    this.timer = setInterval(() => {
      increment()
    }, 500)
  }

  componentWillUnmount() {
    clearInterval(this.timer)
  }

  render() {
    const { count } = this.props
    console.log('render Counter')

    return (
      <div>
        <h3>Count: {count}</h3>
      </div>
    )
  }
}

const mapStateToProps = state => ({ count: state.counterReducer.count })
const mapDispatchToProps = { increment }

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(Counter)
