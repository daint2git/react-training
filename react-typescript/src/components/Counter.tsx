import React, { ReactElement, useState, useEffect } from 'react'

interface Props {
  initialCount: number
  onMessageShow: (count: number) => void
  check?: boolean
}

const Counter = ({ initialCount, onMessageShow }: Props): ReactElement => {
  const [count, setCount] = useState(initialCount)

  useEffect((): void => {
    onMessageShow(count)
    console.log(initialCount)
  })

  return (
    <div>
      <h2>Count {count}</h2>
      <button
        type="button"
        onClick={(): void => {
          setCount(count + 1)
        }}
      >
        +
      </button>
      <button type="button" onClick={(): void => setCount(count - 1)}>
        -
      </button>
    </div>
  )
}

interface ParentCounterProps {
  initialCount: number
}

const ParentCounter = ({ initialCount }: ParentCounterProps): ReactElement => {
  const handleMessageShow = (count: number): void => {
    console.log(`This is ${count}`)
  }

  return <Counter initialCount={initialCount} onMessageShow={handleMessageShow} />
}

export default ParentCounter
