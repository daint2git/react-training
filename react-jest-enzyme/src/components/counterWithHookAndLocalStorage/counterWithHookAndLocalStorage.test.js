import React from 'react'
import { render, fireEvent } from 'react-testing-library'
import Counter from '.'

describe('CounterWithHooks', () => {
  afterEach(() => {
    window.localStorage.removeItem('count')
  })

  test('reads and writes to localStorage (count is undefined)', () => {
    const { container } = render(<Counter />)
    const incrementBtn = container.querySelector('button[name="increment"]')
    fireEvent.click(incrementBtn)
    expect(window.localStorage.getItem('count')).toEqual('1')
  })

  test('reads and writes to localStorage (increment)', () => {
    window.localStorage.setItem('count', 5)
    const { container } = render(<Counter />)
    const incrementBtn = container.querySelector('button[name="increment"]')
    fireEvent.click(incrementBtn)
    expect(window.localStorage.getItem('count')).toEqual('6')
  })

  test('reads and writes to localStorage (decrement)', () => {
    window.localStorage.setItem('count', 5)
    const { container } = render(<Counter />)
    const decrementBtn = container.querySelector('button[name="decrement"]')
    fireEvent.click(decrementBtn)
    expect(window.localStorage.getItem('count')).toEqual('4')
  })

  test('step is 5 + reads and writes to localStorage', () => {
    window.localStorage.setItem('count', 5)
    const { container } = render(<Counter step={5} />)
    const incrementBtn = container.querySelector('button[name="increment"]')
    fireEvent.click(incrementBtn)
    expect(window.localStorage.getItem('count')).toEqual('10')
  })

  test('matches the snapshot', () => {
    const { asFragment } = render(<Counter />)
    expect(asFragment()).toMatchSnapshot()
  })

  test('matches the snapshot when click increment button', () => {
    const { container } = render(<Counter />)
    const incrementBtn = container.querySelector('button[name="increment"]')
    fireEvent.click(incrementBtn)
    const { asFragment } = render(<Counter />)
    expect(asFragment()).toMatchSnapshot()
  })
})
