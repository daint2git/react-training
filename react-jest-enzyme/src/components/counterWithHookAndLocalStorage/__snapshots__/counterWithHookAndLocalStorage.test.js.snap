// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CounterWithHooks matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="counter"
  >
    <p>
      Current count: 0
    </p>
    <button
      name="increment"
    >
      increment
    </button>
    <button
      name="decrement"
    >
      decrement
    </button>
  </div>
</DocumentFragment>
`;

exports[`CounterWithHooks matches the snapshot when click increment button 1`] = `
<DocumentFragment>
  <div
    class="counter"
  >
    <p>
      Current count: 1
    </p>
    <button
      name="increment"
    >
      increment
    </button>
    <button
      name="decrement"
    >
      decrement
    </button>
  </div>
</DocumentFragment>
`;
